projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-coauthor.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * REST API endpoint for the CoAuthor blocks.
+ *
+ * @package automattic/jetpack
+ * @since $$next-version$$
+ */
+
+/**
+ * Class WPCOM_REST_API_V2_Endpoint_Coauthor
+ */
+class WPCOM_REST_API_V2_Endpoint_Coauthor extends WP_REST_Controller {
+	/**
+	 * Namespace prefix.
+	 *
+	 * @var string
+	 */
+	public $namespace = 'wpcom/v2';
+
+	/**
+	 * Endpoint base route.
+	 *
+	 * @var string
+	 */
+	public $rest_base = 'coauthor';
+
+	/**
+	 * WPCOM_REST_API_V2_Endpoint_Coauthor constructor.
+	 */
+	public function __construct() {
+		$this->is_wpcom = false;
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$this->is_wpcom = true;
+		} elseif ( ! ( new Automattic\Jetpack\Status\Host() )->is_woa_site() ) {
+			// If this is not an atomic site, we want to bail and not even load the endpoint for now.
+			return;
+		}
+
+		if ( ! class_exists( 'Jetpack_Coauthor_Helper' ) ) {
+			require_once JETPACK__PLUGIN_DIR . '_inc/lib/class-jetpack-coauthor-helper.php';
+		}
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/completions',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'request_gpt_completion' ),
+					'permission_callback' => array( 'Jetpack_Coauthor_Helper', 'get_status_permission_check' ),
+				),
+				'args' => array(
+					'content' => array( 'required' => true ),
+					'token'   => array( 'required' => false ),
+				),
+			)
+		);
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/images/generations',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'request_dalle_generation' ),
+					'permission_callback' => array( 'Jetpack_Coauthor_Helper', 'get_status_permission_check' ),
+				),
+				'args' => array(
+					'prompt' => array( 'required' => true ),
+					'token'  => array( 'required' => false ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get completions for a given text.
+	 *
+	 * @param  WP_REST_Request $request The request.
+	 */
+	public function request_gpt_completion( $request ) {
+		return Jetpack_Coauthor_Helper::get_gpt_completion( $request['content'] );
```

Comment:
I don't really see the use of `Jetpack_Coauthor_Helper`. It is basically implementing the endpoint...

Reply:
It does keep things strictly related to defining the routes here, but I could see it both ways for sure.